usecases/cluster/transactions_write.go

```diff
@@ -466,32 +466,69 @@ func (c *TxManager) IncomingCommitTransaction(ctx context.Context,
 	c.ongoingCommits.Add(1)
 	defer c.ongoingCommits.Done()
 
+	// requires locking because it accesses c.currentTransaction
+	txCopy, err := c.incomingCommitTxValidate(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	// cannot use locking because of risk of deadlock, see comment inside method
+	if err := c.incomingTxCommitApplyCommitFn(ctx, txCopy); err != nil {
```

Comment:
We don't update txCopy with thre returned value from `incomingTxCommitApplyCommitFn`, which mean we then use the original tx for cleanup instead of the updated one. Not sure if critical but as we are doing it between `incomingCommitTxValidate` and `incomingTxCommitApplyCommitFn` maybe it makes sense to keep doing it ?

Reply:
The cleanup is purely ID-based (it sets the `.currentTransaction` to `nil` and deletes by id from the store). So I _think_ it should have no real impact. Maybe to make that clear it would be cleaner to not pass the whole tx, but only the ID?

Reply:
If it's using only the ID then it should be OK I think.